src/MICore/Debugger.cs

```diff
@@ -1161,15 +1136,14 @@ private void HandleThreadGroupExited(Results results)
                 {
                     if (_debuggeePids.Remove(threadGroupId))
                     {
-                        if (_debuggeePids.Count == 0)
-                            ProcessState = ProcessState.Exited;
+                        isThreadGroupEmpty = _debuggeePids.Count == 0;
                     }
                 }
             }
 
-            if (ProcessState == ProcessState.Exited)
+            if (isThreadGroupEmpty)
             {
-                OnDebuggerProcessExit(null);
+                ScheduleStdOutProcessing(@"*stopped,reason=""exited""");
```

Comment:
Does the HandleThreadGroupExited path still lead to Close() eventually being invoked without the direct call to OnDebuggerProcessExit?